core/src/test/java/org/infinispan/test/AbstractInfinispanTest.java

```diff
@@ -72,6 +92,66 @@
 @TestSelector(interceptors = AbstractInfinispanTest.OrderByInstance.class)
 public abstract class AbstractInfinispanTest {
 
+   static {
+      BlockHound.builder().with(new RxJava2Integration()).with(builder -> {
+         builder.markAsBlocking(CacheImpl.class, "size", "()I");
+         builder.markAsBlocking(CacheImpl.class, "size", "(J)I");
+         builder.markAsBlocking(CacheImpl.class, "containsKey", "(Ljava/lang/Object;)Z");
+         builder.markAsBlocking(CacheImpl.class, "get", "(Ljava/lang/Object;)Ljava/lang/Object;");
+         builder.markAsBlocking(CacheImpl.class, "put", "(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;");
+
+         builder.nonBlockingThreadPredicate(current -> current.or(thread -> thread instanceof NonBlockingThread));
+         // SecureRandom reads from a socket
+         builder.allowBlockingCallsInside(SecureRandom.class.getName(), "nextBytes");
+         // UDP calls PlainDatagramSocketImpl#send which can block if OS buffer is full - but we just ignore
+         builder.allowBlockingCallsInside(UDP.class.getName(), "_send");
+         builder.allowBlockingCallsInside(FlowControl.class.getName(), "adjustCredit");
+         // Random JGroups stuff
+         builder.allowBlockingCallsInside(Table.class.getName(), "add");
+         // Just assume all the thread pools don't block - not we check the rejected handler below though
+         builder.allowBlockingCallsInside(ThreadPoolExecutor.class.getName(), "getTask");
+         builder.allowBlockingCallsInside(ThreadPoolExecutor.class.getName(), "processWorkerExit");
+         builder.allowBlockingCallsInside(ThreadPoolExecutor.class.getName(), "execute");
+         builder.allowBlockingCallsInside(ScheduledThreadPoolExecutor.class.getName(), "schedule");
+
+         // TODO: This can actually block us for a bit, we should really remove this!!!
```

Comment:
This shouldn't be true anymore afaik.

Reply:
I need to update the comment, this was fixed in https://github.com/infinispan/infinispan/commit/df4e35761f1ac2016059b2265882d77350e7532a#diff-9d071c3a5e71e018abe6e735ec5a62fb. The problem prior was that the exclusive lock was held while purging store segments. This has been since fixed, so this method shouldn't block at all really.